tests/vmi_tpm_test.go

```diff
@@ -85,3 +101,182 @@ var _ = Describe("[sig-compute]vTPM", decorators.SigCompute, func() {
+				&expect.BExp{R: "MYSECRET"},
+			}, 300)).To(Succeed(), "the state of the TPM did not persist")
+		}
+
+		DescribeTable("[Serial]should persist TPM secrets across", Serial, func(op1, op2 string) {
+			By("Setting the backend storage class to the default for RWX FS or skip if missing")
+			storageClass, exists := libstorage.GetRWXFileSystemStorageClass()
+			Expect(exists).To(BeTrue(), "No RWX FS storage class found")
+			kv := util.GetCurrentKv(virtClient)
+			kv.Spec.Configuration.VMStateStorageClass = storageClass
+			tests.UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+			By("Creating a VM with persistent TPM enabled")
+			vmi := tests.NewRandomFedoraVMI()
+			vmi.Namespace = util.NamespaceTestDefault
+			vmi.Spec.Domain.Devices.TPM = &v1.TPMDevice{
+				Persistent: pointer.BoolPtr(true),
+			}
+			vm := tests.NewRandomVirtualMachine(vmi, true)
+			vm, err = virtClient.VirtualMachine(util.NamespaceTestDefault).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Waiting for the VM to start")
+			Eventually(func() error {
+				vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				return err
+			}, 300*time.Second, 1*time.Second).Should(Succeed())
+			libwait.WaitForSuccessfulVMIStartWithTimeout(vmi, 60)
+
+			By("Logging in as root")
+			err = console.LoginToFedora(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Storing a secret into the TPM")
+			// https://www.intel.com/content/www/us/en/developer/articles/code-sample/protecting-secret-data-and-keys-using-intel-platform-trust-technology.html
+			// Not sealing against a set of PCRs, out of scope here, but should work with a carefully selected set (at least PCR1 was seen changing accross reboots)
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "tpm2_createprimary -Q --hierarchy=o --key-context=prim.ctx\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "echo MYSECRET | tpm2_create --hash-algorithm=sha256 --public=seal.pub --private=seal.priv --sealing-input=- --parent-context=prim.ctx\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "tpm2_load -Q --parent-context=prim.ctx --public=seal.pub --private=seal.priv --name=seal.name --key-context=seal.ctx\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "tpm2_evictcontrol --hierarchy=o --object-context=seal.ctx 0x81010002\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "tpm2_unseal -Q --object-context=0x81010002\n"},
+				&expect.BExp{R: "MYSECRET"},
+			}, 300)).To(Succeed(), "failed to store secret into the TPM")
+
+			if op1 == "migrate" {
+				migrateVMI(vmi)
+			} else if op1 == "restart" {
+				restartVM(vm)
+			}
+
+			checkTPM(vmi)
+
+			if op2 == "migrate" {
+				migrateVMI(vmi)
+			} else if op2 == "restart" {
+				restartVM(vm)
+			}
+
+			checkTPM(vmi)
```

Comment:
Instead of `op1` and `op2` you can switch using `ops ...string` and then loop through it with:
```suggestion
			for _, op := range ops{
				switch op {
				case "migrate":
					migrateVMI(vmi)
				case "restart":
					restartVM(vm)
				}
				checkTPM(vmi)
			}
```